Question: should google-cloud-spanner-jdbc use the Libraries BOM?

While troubleshooting https://github.com/googleapis/java-spanner-jdbc/pull/251#issuecomment-721256007, I found the pom.xml of this repository imports the Libraries BOM. (Since when? The PR #133 in May 2020 added the Libraries BOM in this project's pom.xml through dependencyManagement section's import. It seems that the PR intended to make it easy to specify the versions of multiple libraries.)

However, the Libraries BOM includes this repository's library ([`com.google.cloud:google-cloud-spanner-jdbc`](https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/snapshot/com.google.cloud_google-cloud-spanner-jdbc_1.17.0.html)) through google-cloud-bom. It's a circular dependency. This circular dependency makes dependency troubleshooting difficult in https://github.com/googleapis/java-spanner-jdbc/pull/251 .

Shall we use the shared dependency BOM only and remove the libraries BOM from the dependency management section?

Here is removal of the library versions from pom.xml: https://github.com/googleapis/java-spanner-jdbc/pull/133/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R59

- grpc.version
- api-client.version
- google.core.version
- gax.version
- http-client-bom.version
- guava.version
- google.common-protos.version
- protobuf.version
- google.api-common.version
- spanner.version
- google.auth.version

Simply removing the libraries-bom generates the following errors:

```
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project com.google.cloud:google-cloud-spanner-jdbc:1.17.3-SNAPSHOT (/Users/suztomo/java-spanner-jdbc/pom.xml) has 6 errors
[ERROR]     'dependencies.dependency.version' for com.google.cloud:google-cloud-spanner:jar is missing. @ line 93, column 17
[ERROR]     'dependencies.dependency.version' for com.google.api.grpc:proto-google-cloud-spanner-v1:jar is missing. @ line 137, column 17
[ERROR]     'dependencies.dependency.version' for com.google.api.grpc:grpc-google-cloud-spanner-v1:jar is missing. @ line 153, column 17
[ERROR]     'dependencies.dependency.version' for com.google.cloud:google-cloud-spanner:test-jar is missing. @ line 147, column 17
[ERROR]     'dependencies.dependency.version' for com.google.api.grpc:grpc-google-cloud-spanner-admin-instance-v1:jar is missing. @ line 158, column 17
[ERROR]     'dependencies.dependency.version' for com.google.api.grpc:grpc-google-cloud-spanner-admin-database-v1:jar is missing. @ line 163, column 17
[ERROR] 

```

Continuing.

--- a/pom.xml
+++ b/pom.xml
@@ -70,8 +70,8 @@
     <dependencies>
       <dependency>
         <groupId>com.google.cloud</groupId>
-        <artifactId>libraries-bom</artifactId>
-        <version>12.0.0</version>
+        <artifactId>google-cloud-spanner-bom</artifactId>
+        <version>2.0.2</version>
         <type>pom</type>
         <scope>import</scope>
       </dependency>

